In Assert.php line 2042: Expected a value other than null.

| Question    | Answer
| ------------| ---------------
| Infection version | 0.18.0
| Test Framework version | PHPUnit
| PHP version | 7.4.10
| Platform    | windows-github/actions
| Github Repo | https://github.com/yiisoft/active-record/runs/1270725926?check_suite_focus=true

```php
Processing source code files: 0
........U.........T..........T...............UU..U   (   50)
UU............................U...........U.......   (  100)
.......................U...U...UUUUUUUUUUUUUUUUUUU   (  150)
UUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUU..   (  200)
...............U.....................UU...........   (  250)
....M................UU...........................   (  300)
......M.............................U.......U....U   (  350)
...............U.........U......U.........U.....UU   (  400)

In Assert.php line 2042:
                                     
  Expected a value other than null.  
                                     

run [--test-framework TEST-FRAMEWORK] [--test-framework-options TEST-FRAMEWORK-OPTIONS] [-j|--threads THREADS] [--only-covered] [-s|--show-mutations] [--no-progress] [--force-progress] [-c|--configuration CONFIGURATION] [--coverage COVERAGE] [--mutators MUTATORS] [--filter FILTER] [--formatter FORMATTER] [--min-msi MIN-MSI] [--min-covered-msi MIN-COVERED-MSI] [--log-verbosity LOG-VERBOSITY] [--initial-tests-php-options INITIAL-TESTS-PHP-OPTIONS] [--skip-initial-tests] [--ignore-msi-with-no-mutations] [--debug] [--dry-run]

UUUSS.SSS
Error: Process completed with exit code 1.
```

--- a/src/Mutator/Removal/ArrayItemRemoval.php
+++ b/src/Mutator/Removal/ArrayItemRemoval.php
@@ -41,6 +41,7 @@ use Infection\Mutator\Definition;
 use Infection\Mutator\GetConfigClassName;
 use Infection\Mutator\GetMutatorName;
 use Infection\Mutator\MutatorCategory;
+use Infection\PhpParser\Visitor\ParentConnector;
 use function min;
 use PhpParser\Node;
 use PhpParser\Node\Expr\ArrayItem;
@@ -119,7 +120,22 @@ TXT
 
     public function canMutate(Node $node): bool
     {
-        return $node instanceof Node\Expr\Array_ && count($node->items);
+        if (!$node instanceof Node\Expr\Array_) {
+            return false;
+        }
+
+        if (!count($node->items)) {
+            return false;
+        }
+
+        $parent = ParentConnector::findParent($node);
+
+        // Arrays to the left of an assignments are not arrays but lists.
+        if ($parent instanceof Node\Expr\Assign && $parent->var === $node) {
+            return false;
+        }
+
+        return true;
     }
 
     /**

--- a/src/Mutator/Removal/ArrayItemRemoval.php
+++ b/src/Mutator/Removal/ArrayItemRemoval.php
@@ -124,7 +124,7 @@ TXT
             return false;
         }
 
-        if (!count($node->items)) {
+        if (count($node->items) === 0) {
             return false;
         }
 

--- a/src/Mutator/Removal/ArrayItemRemoval.php
+++ b/src/Mutator/Removal/ArrayItemRemoval.php
@@ -124,7 +124,7 @@ TXT
             return false;
         }
 
-        if (count($node->items) === 0) {
+        if ($node->items === []) {
             return false;
         }
 

--- a/tests/phpunit/Mutator/Removal/ArrayItemRemovalTest.php
+++ b/tests/phpunit/Mutator/Removal/ArrayItemRemovalTest.php
@@ -102,5 +102,17 @@ final class ArrayItemRemovalTest extends BaseMutatorTestCase
             ],
             ['remove' => 'all', 'limit' => 1],
         ];
+
+        yield 'It does not mutate lists with missing elements' => [
+            '<?php [, $a] = [];',
+        ];
+
+        yield 'It does not mutate lists with one element' => [
+            '<?php [$a] = [];',
+        ];
+
+        yield 'It does not mutate lists with any number of elements' => [
+            '<?php [$a, $b] = [];',
+        ];
     }
 }

